packages/varlock-github-action/README.md

```diff
@@ -0,0 +1,236 @@
+# Varlock GitHub Action
+
+A GitHub Action that loads and validates environment variables using [varlock](https://github.com/dmno-dev/varlock). This action automatically detects varlock installations or .env.schema/.env.* files and loads validated environment variables into the GitHub Actions environment.
+
+## Features
+
+- 🔍 **Automatic Detection**: Checks for varlock installation or compatible env files
+- 📦 **Auto-Installation**: Installs varlock if not found
+- 🔒 **Schema Validation**: Validates environment variables against your schema
+- 📋 **Summary Output**: Provides detailed summaries of loaded variables
+- ⚙️ **Flexible Configuration**: Supports different output formats and environments
+
+## Usage
+
+### Basic Usage
+
+```yaml
+name: Load Environment Variables
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Load environment variables
+        uses: dmno-dev/varlock-github-action@v1
+      
+      - name: Use loaded variables
+        run: |
+          echo "Database URL: $DATABASE_URL"
+          echo "API Key: $API_KEY"
+```
+
+### With Custom Configuration
+
+```yaml
+name: Load Environment Variables
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Load environment variables
+        uses: dmno-dev/varlock-github-action@v1
+        with:
+          working-directory: './config'
+          environment: 'production'
+          show-summary: 'true'
+          fail-on-error: 'true'
+          output-format: 'env'
+      
+      - name: Use loaded variables
+        run: |
+          echo "Environment loaded successfully"
+```
+
+## Inputs
+
+| Input | Description | Required | Default |
+|-------|-------------|----------|---------|
+| `working-directory` | Directory containing @env-spec files | No | `.` |
+| `environment` | Environment to load (e.g., production, development) | No | - |
+| `show-summary` | Show a summary of loaded environment variables | No | `true` |
+| `fail-on-error` | Fail the action if validation errors are found | No | `true` |
+| `output-format` | Format for environment variable output (env, json) | No | `env` |
```

Comment:
how is the output used?

Reply:
that can be reworked a bit similar to the dmno action where we either output individual env vars or one big json blob, it's more relevant for multi-stage workflows where the env vars don't persist